### PR TITLE
[VL] Enable GlutenCoalesceShufflePartitionsSuite.

### PR DESCRIPTION
**Pull request by lviiii**
## What changes were proposed in this pull request?

Rewrite UTs to check columnar operators instead of Vanilla Spark operators.

## How was this patch tested?

Passed UT and CI.

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

File: gluten-ut/spark32/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala [hunk 1]
```diff
@@ -356,26 +359,4 @@ class VeloxTestSettings extends BackendTestSettings {
   enableSuite[GlutenDataSourceV2Suite]
     .exclude("columnar batch scan implementation")
     .exclude("partitioning reporting")
-  enableSuite[GlutenApproxCountDistinctForIntervalsQuerySuite]
```
Comment (PHILO-HE) [inline gluten-ut/spark32/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala:359]: These suites were unintentionally deleted?

**Comment by FelixYBW** [comment]
@rui-mo can you follow up?

**Comment by PHILO-HE** [comment]
> @rui-mo can you follow up?

I will take it to continue the fix.

**Comment by PHILO-HE** [comment]
We have another PR to cover this fix, with some code ported there. Thanks @lviiii for your work!
https://github.com/oap-project/gluten/pull/1007

**Pull request closed without merging**